cmd/lifecycle/exporter.go

```diff
@@ -205,6 +227,35 @@ func (e *exportCmd) registryImages() []string {
 	return registryImages
 }
 
+func (e *exportCmd) populateRunImageRefIfNeeded() error {
+	if !e.supportsRunImage() {
+		if e.analyzedMD.RunImage == nil || e.analyzedMD.RunImage.Reference == "" {
+			return errors.New("run image not found in analyzed metadata")
+		}
+		e.runImageRef = e.analyzedMD.RunImage.Reference
+	} else if e.runImageRef == "" {
+		var err error
+		e.runImageRef, err = e.stackMD.BestRunImageMirror(e.targetRegistry)
+		if err != nil {
+			return errors.New("-run-image is required when there is no stack metadata available")
+		}
+	}
+	return nil
+}
+
+func (e *exportCmd) validateRunImageInput() error {
+	switch {
```

Comment:
Maybe it's worth adding the relevant image names to the error messages in this switch-case.

Reply:
That feels a bit messy to me. I'm not sure having the image names would add much, as the command invocation is already known. I kind of like the terse error message...

Reply:
mmmm...I don't have strong opinion on this but I just thought that printing a clear message with as much information as we can would be useful in case of an error.
If you feel differently, please feel free to mark this conversation as resolved.

Reply:
If there's no strong opinion, I'll mark as resolved for now to avoid making more changes :)